core/src/main/java/fr/sncf/osrd/api/PathfindingResult.java

```diff
@@ -0,0 +1,258 @@
+package fr.sncf.osrd.api;
+
+import com.squareup.moshi.Json;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import fr.sncf.osrd.infra.Infra;
+import fr.sncf.osrd.infra.InvalidInfraException;
+import fr.sncf.osrd.infra.OperationalPoint;
+import fr.sncf.osrd.infra.routegraph.Route;
+import fr.sncf.osrd.infra.routegraph.RouteLocation;
+import fr.sncf.osrd.infra.trackgraph.TrackSection;
+import fr.sncf.osrd.railjson.schema.common.RJSObjectRef;
+import fr.sncf.osrd.railjson.schema.infra.RJSRoute;
+import fr.sncf.osrd.railjson.schema.infra.RJSTrackSection;
+import fr.sncf.osrd.train.TrackSectionRange;
+import fr.sncf.osrd.utils.PointValue;
+import fr.sncf.osrd.utils.TrackSectionLocation;
+import fr.sncf.osrd.utils.geom.LineString;
+import fr.sncf.osrd.utils.graph.EdgeDirection;
+import fr.sncf.osrd.utils.graph.path.BasicPathNode;
+import fr.sncf.osrd.utils.graph.path.FullPathArray;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PathfindingResult {
+    @Json(name = "route_paths")
+    public final List<RoutePathResult> routePaths = new ArrayList<>();
+    @Json(name = "path_waypoints")
+    public final List<PathWaypointResult> pathWaypoints = new ArrayList<>();
+
+    // make them private final
+    public LineString geographic = null;
+    public LineString schematic = null;
+
+    /**
+     * pathfindingResultMake is used to create a class PathfindingResult with a good format to be sent to the middleware
+     * during this function, we add geometric information in the PathfindingResult
+     * @param finalPathsToGoal contains the routes that are use after the pathfinding
+     * @param infra current infra
+     * @return A pathfindingResult object ready to be sent to the middleware
+     */
```

Comment:
We have as a principle to avoid making docs that paraphrase the code.
The `@return` and `@param infra` does not seem usefull.